### PR TITLE
feat: Add dark mode in Gatsby site

### LINKED ISSUE #185: Add dark mode support for Gatsby site
### What feature do you want to see added?

To add dark mode support to Gatsby site like what we have done for https://www.jenkins.io/. 

### Upstream changes

N/A

### PR DESCRIPTION
**Pull request by saurabhraghuvanshii**
Add dark mode for the Gatsby site

Closes #185 

Screenshot 2024-12-01 002135
Screenshot 2024-12-01 002152
Screenshot 2024-12-01 002218

### CODE REVIEW

File: site/src/css/authorpost.module.css [hunk 1]
```diff
@@ -120,4 +120,4 @@ li {
     align-items: center;
     justify-content: center;
     height: 15rem;
-}
+}
```
Comment (krisstern) [inline site/src/css/authorpost.module.css:123]:
```suggestion
}

```

File: site/src/css/blogpost.module.css [hunk 2]
```diff
@@ -134,3 +135,22 @@ li {
...
+    }
+    
+    .blogtitle {
+        color: hsl(240, 15%, 95%);
+    }
+    
+    .blogteaser {
+        color: hsl(240, 15%, 95%);
+    }
+}
```
Comment (krisstern) [inline site/src/css/blogpost.module.css:156]:
```suggestion
}

```

**Review by krisstern — CHANGES_REQUESTED** [review]

**Comment by saurabhraghuvanshii** [comment]
@krisstern sir , is any more changes required?

**Review by krisstern — APPROVED** [review]
Thanks @saurabhraghuvanshii for the PR! It looks good to me

**Comment by saurabhraghuvanshii** [comment]
@krisstern welcome sir , sir I can make it responsive.

**Pull request merged**